planner/core/find_best_task.go

```diff
@@ -500,11 +500,18 @@ func (ds *DataSource) getIndexCandidate(path *util.AccessPath, prop *property.Ph
 	// When the prop is empty or `all` is false, `isMatchProp` is better to be `false` because
 	// it needs not to keep order for index scan.
 	if !prop.IsEmpty() && all {
+		eqCondCount := 0
+		for _, ran := range path.Ranges {
+			// Some expressions such as `a >= 2 && a <= 2` can also convert to a equal condition like `a = 2`.
+			if ran.IsPoint(ds.ctx.GetSessionVars().StmtCtx) {
+				eqCondCount++
```

Comment:
Would it be better to add this check / modification at `detachCNFCondAndBuildRangeForIndex`?